test/TestProviderRound.spec.js

```diff
@@ -5,40 +5,56 @@ require('truffle-test-utils').init();
+      await providerRound.provider(pricePerStorageMineral, pricePerComputeMineral, blockRewardCut, feeShare, {from: providerAddress});
+  }
+
   describe('provider', () => {
 
     before(async () => {
       providerRound = await ProviderRound.deployed();
+      contractAddress = providerRound.address;
+      for(let i = 0; i < 5; i++) {
+        await providerRound.mint(accounts[i], 1000, {from: accounts[0]});
+      }
+      await providerRound.setMaxNumberOfProviders(PROVIDER_POOL_SIZE);
     });
 
     beforeEach(async () => {
       await blockMiner.mineUntilEndOfElectionPeriod(providerRound);
       await providerRound.initializeRound();
     });
 
+    it('should fail if the provider does not bond some tokens on himself first', async () => {
+      await assertFail( providerRound.provider(22, 10, 1, 25, {from: accounts[0]}) );
+    });
+
+    it('should initially set totalBondedAmount to the amount the provider bonded to himself', async () => {
+      await approveBondProvider(22, 10, 1, 25, 42, accounts[0]);
+      const provider = await providerRound.providers.call(accounts[0]);
+      assert.equal(42, provider[5]); // [5] is totalBondedAmount
+    });
+
     it("should register a provider's parameters", async () => {
-      await providerRound.provider(22, 10, 1, 25, {from: accounts[0]});
-      await providerRound.provider(10, 20, 2, 35, {from: accounts[1]});
-      const firstProvider = await providerRound.providers.call(accounts[0]);
-      assert.equal(firstProvider[0], PROVIDER_REGISTERED); // [0] is providerStatus
-      assert.equal(firstProvider[1].toNumber(), 22); // [1] is pricePerStorageMineral
-      assert.equal(firstProvider[2].toNumber(), 10); // [2] is pricePerComputeMineral
-      assert.equal(firstProvider[3].toNumber(), 1);  // [3] is blockRewardCut
-      assert.equal(firstProvider[4].toNumber(), 25); // [4] is feeShare
-      const secondProvider = await providerRound.providers.call(accounts[1]);
-      assert.equal(secondProvider[0], PROVIDER_REGISTERED); // [0] is providerStatus
-      assert.equal(secondProvider[1].toNumber(), 10); // [1] is pricePerStorageMineral
-      assert.equal(secondProvider[2].toNumber(), 20); // [2] is pricePerComputeMineral
-      assert.equal(secondProvider[3].toNumber(), 2);  // [3] is blockRewardCut
-      assert.equal(secondProvider[4].toNumber(), 35); // [4] is feeShare
+      await approveBondProvider(10, 20, 2, 35, 1, accounts[1]);
+      const provider = await providerRound.providers.call(accounts[1]);
+      assert.equal(provider[0], PROVIDER_REGISTERED); // [0] is providerStatus
+      assert.equal(provider[1].toNumber(), 10); // [1] is pricePerStorageMineral
+      assert.equal(provider[2].toNumber(), 20); // [2] is pricePerComputeMineral
+      assert.equal(provider[3].toNumber(), 2);  // [3] is blockRewardCut
+      assert.equal(provider[4].toNumber(), 35); // [4] is feeShare
     });
 
     it('should fail with invalid parameters', async() => {
```

Comment:
Are we only asserting whole numbers here or should we be testing for failure on decimals?

Reply:
Mmmh, you're right. Currently it only works with integers on the side on the contract. Will add that to the backlog